contracts/core/Dispatcher.sol

```diff
@@ -87,26 +87,31 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
             revert IBCErrors.invalidCounterPartyPortId();
         }
 
-        if (_isChannelOpenTry(counterparty)) {
+        if (Ibc._isChannelOpenTry(counterparty)) {
             consensusStateManager.verifyMembership(
                 proof,
-                channelProofKey(local.portId, local.channelId),
-                channelProofValue(ChannelState.TRY_PENDING, ordering, local.version, connectionHops, counterparty)
+                Ibc.channelProofKey(local.portId, local.channelId),
+                Ibc.channelProofValue(ChannelState.TRY_PENDING, ordering, local.version, connectionHops, counterparty)
             );
         }
 
-        string memory selectedVersion =
-            portAddress.onOpenIbcChannel(local.version, ordering, feeEnabled, connectionHops, counterparty);
-
-        emit OpenIbcChannel(
-            address(portAddress),
-            selectedVersion,
-            ordering,
-            feeEnabled,
-            connectionHops,
-            counterparty.portId,
-            counterparty.channelId
-        );
+        string memory selectedVersion;
+        try portAddress.onOpenIbcChannel(local.version, ordering, feeEnabled, connectionHops, counterparty) returns (
```

Comment:
Do we have a test the covers the case where `portAddress` is not a real contract? Will this try/catch "catch" that situation?

Reply:
yes thank you so much for raising this! I thought try/catch protected against EOA calls, but it actually doesn't. 

So I refactored the code to use low-level calls, and that should support all cases (e.g. reverting contracts, contracts without the handlers, and EOA) and the tests for these and it should be working now